### DISCUSSION TITLE
DCC++ EX / JMRI Lights as Outputs

### DISCUSSION THREAD

**Discussion by ianmidd** [DCC-EX/CommandStation-EX#247] [Q&A]
Apologies for being a noob. I have success with DCC++ EX and JMRI and have added Sensors using Arduino Mega pins (for testing). I have tried on and off for two weeks now to setup Test outputs on pins 3 and 13.

In the DCC commands it acknowledges the Pin 3 and Pin 13 assignment as Outputs as it does for Sensor Inputs on Pins 52 and 53.

I sense that problem lies in the "Hardware Address" for Pins 3 and 13. I have assumed these are just 3 and 13 but the DCC message sent is referring to a decoder address.

Any help appreciated as I am about to buy a gun and shoot myself.
[New Text Document (2).txt](https://github.com/DCC-EX/CommandStation-EX/files/8881661/New.Text.Document.2.txt)

**Comment by Asbelos** [comment]
Pins 51 52 and 53 have special meaning on a mega. You need to avoid them.
Pins 3 and 13 are normally used to communicate between the mega and the motor shield so will go berserk if you try and sens or output to them. 

These Pins depend on your hardware setup snd configuration... see MotorDrivers.h for a discussion on this.

**Reply by ianmidd** [reply]
Sorry Asbelos, I don’t understand that. Pins 52/53 are working fine as Sensors and pin 13 is reserved for the internal LED or so I thought. 

Let’s say I use pins 49/48 as lights. I thought the correct ID in the JMRI would be 49 & 48 respectively.

**Comment by ianmidd** [comment]
Turns out this is all linked to ISCLOCKRUNNING. 

Regards,

Ian
Sent from my iPhone 
+44 7768 215798
